src/test/java/com/opipo/codechallenge/service/TransactionAmountServiceTest.java

```diff
@@ -0,0 +1,112 @@
+                Arguments.of(45d,
+                        new TransactionEntity[] {buildTransactionEntity(10d, 0.1d), buildTransactionEntity(11d, 0.2d),
+                                buildTransactionEntity(12d, 0.3d), buildTransactionEntity(13d, 0.4d)}),
+                Arguments.of(4d, new TransactionEntity[] {buildTransactionEntity(23d, 19d)}));
+    }
+
+    private static Stream<Arguments> provideSums() {
+        return Stream.of(Arguments.of(10d, 1d, true), Arguments.of(0d, 10d, true), Arguments.of(-10d, 1d, false),
+                Arguments.of(1d, -10d, false), Arguments.of(null, -1d, false), Arguments.of(1d, null, true),
+                Arguments.of(1d, 1d, true), Arguments.of(1d, -1d, true));
+    }
+
+    private static TransactionEntity buildTransactionEntity(Double amount, Double fee) {
+        TransactionEntity transactionEntity = new TransactionEntity();
+        transactionEntity.setAmount(amount == null ? null : BigDecimal.valueOf(amount));
+        transactionEntity.setFee(fee == null ? null : BigDecimal.valueOf(fee));
+        return transactionEntity;
+    }
+
+    @ParameterizedTest(name = "Given amount {0} and fee {1}, {2} is expected")
+    @MethodSource("provideAmounts")
+    @DisplayName("Given amount and fee, then check the sum")
+    @Execution(ExecutionMode.CONCURRENT)
+    void calculateAmount(final Double amount, final Double fee, final Double result) {
+        assertEquals(BigDecimal.valueOf(result), transactionAmountService.calculateAmount(
+                amount == null ? null : BigDecimal.valueOf(amount), fee == null ? null : BigDecimal.valueOf(fee)));
+    }
+
+    @ParameterizedTest(name = "Given transaction with amount {0} and fee {1}, {2} is expected")
+    @MethodSource("provideAmounts")
+    @DisplayName("Given Transaction with amount and fee, then check the sum")
+    @Execution(ExecutionMode.CONCURRENT)
+    void calculateAmountWithTransaction(final Double amount, final Double fee, final Double result) {
+        assertEquals(BigDecimal.valueOf(result),
+                transactionAmountService.calculateAmount(buildTransactionEntity(amount, fee)));
+    }
+
+    @Test
+    @DisplayName("Given null amount and any fee, IllegalArgumentException is expected")
+    @Execution(ExecutionMode.CONCURRENT)
+    void givenNullAmountThenThrowNullPointer() {
+        assertThrows(IllegalArgumentException.class,
+                () -> transactionAmountService.calculateAmount(null, BigDecimal.ONE));
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideTransactions")
+    @DisplayName("Given transaction stream, then sumarize all")
+    @Execution(ExecutionMode.CONCURRENT)
+    void givenTransactionsThenSumarizeIt(final Double result, final TransactionEntity[] transactionArray) {
+        assertEquals(BigDecimal.valueOf(result), transactionAmountService.sumarize(Arrays.asList(transactionArray)));
+    }
+
+    @ParameterizedTest
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/app/albertinizao/codechallenge/pullRequest?prid=4681930)